frontend/public/components/pod-logs.jsx

```diff
@@ -1,194 +1,79 @@
-    this.setState(this._initialState(nextProps.obj));
   }
 
-  componentWillUnmount() {
-    super.componentWillUnmount();
-    this._endStreaming();
-  }
-
-  _initialState(obj = this.props.obj) {
+  static getDerivedStateFromProps(nextProps, prevState) {
     const newState = {};
-
-    const containers = _.get(obj, 'spec.containers', []);
-    newState.containerNames = _.map(containers, 'name');
-
-    if (!this.state.currentContainer && newState.containerNames.length > 0) {
-      newState.currentContainer = newState.containerNames[0];
-    }
-
-    const currentContainer = newState.currentContainer || this.state.currentContainer;
-    newState.logURL = this._logURL(obj, currentContainer);
-
-    return newState;
-  }
-
-  _logURL(obj, currentContainer) {
-    return resourceURL(PodModel, {
-      ns: _.get(obj, 'metadata.namespace'),
-      name: _.get(obj, 'metadata.name'),
-      path: 'log',
-      queryParams: {
-        container: currentContainer,
-        follow: 'true',
-        tailLines: this._buffer.maxSize
-      }
+    const containers = _.get(nextProps.obj, 'status.containerStatuses', []);
+    newState.containers = _.map(containers, (container) => {
+      return {
+        name: container.name,
+        status: container.state
+      };
     });
-  }
 
-  _selectContainer(index) {
-    this._endStreaming();
-    this._buffer = lineBuffer(1000);
-    const currentContainer = this.state.containerNames[index];
-    this.setState({
-      currentContainer,
-      logURL: this._logURL(this.props.obj, currentContainer),
-      logState: 'loading'
-    }, this._beginStreaming);
-  }
-
-  _updateLogState(newState) {
-    this.setState({
-      logState: newState
-    });
-  }
-
-  _toggleLogState() {
-    this.setState({
-      logState: this.state.logState === 'streaming' ? 'paused' : 'streaming'
-    });
-  }
-
-  _touchLoadTime() {
-    this._touchLoadTimeState();
-    this._loadTime = Date.now();
-  }
-
-  // separate function so that it can be throttled
-  _touchLoadTimeState() {
-    this.setState({
-      loadTime: this.state.loadTime + 1
-    });
-  }
-
-  _endStreaming() {
-    this._stream.abort();
-    clearTimeout(this._pendingReload);
-  }
-
-  _resetPendingReload() {
-    const sinceLastLoad = Date.now() - this._loadTime;
-    const wait = Math.max(0, (1000 * 5) - sinceLastLoad);
-    this._pendingReload = setTimeout(() => {
-      if (this.state.logState === 'paused') {
-        // don't reset stream if the user paused the stream
-        this._touchLoadTime();
-        this._resetPendingReload();
-        return;
-      }
-      this._beginStreaming();
-    }, wait);
-  }
+    if (_.isEmpty(prevState.currentContainer) && newState.containers.length > 0) {
+      newState.currentContainer = newState.containers[0];
+    }
 
-  _loadStarted() {
-    this.setState({
-      logState: 'streaming'
-    });
-  }
+    const currentContainer = newState.currentContainer || prevState.currentContainer;
```

Comment:
This code is hard to follow:
```javascript
  if (_.isEmpty(prevState.currentContainer) && newState.containers.length > 0) {
      newState.currentContainer = newState.containers[0];
 }

const currentContainer = newState.currentContainer || prevState.currentContainer;
```

I think you can simplify it to the following (if you store `eof` in the containers instead of `status`):

```javascript
newState.currentContainer = prevState.currentContainer || newState.containers[0];
```